src/sagas/projects.js

```diff
@@ -87,6 +92,38 @@ export function* updateProjectSource() {
   yield* saveCurrentProject();
 }
 
+export async function importBeautify() {
+  return retryingFailedImports(
+    () => import(
+      /* webpackChunkName: "mainAsync" */
+      'js-beautify',
+    ),
+  );
+}
+
+export function* loadAndBeautifyProjectSource() {
+  const beautify = yield call(importBeautify);
```

Comment:
i was following importing `validations` in `errors.js`

Reply:
Got it—yeah it’s kind of a judgement call in each case, but in general I try to ask the question “what is the biggest chunk I can bite off with an async load?”—so in this case the `formatter` module itself can be loaded asynchronously with no major downsides that I can think of, and I think would make the interface a little simpler since you no longer need to pass in a `beautify` instance?

Reply:
i don't fully understand how a static load works. i understand the entire point of this is to have a dynamic import, but when is the code to load Beautify actually run?